[ FL-gradient quantization] Adding QNN unpack feature

kwanghoon-meta opened the pull request:
Summary: We are trying to add a new feature for quantized gradient computation which enables backward() function for QNNPACK

Test Plan: buck2 test //caffe2/test/quantization:quantization -- test_qlinear_qnnpack_free_memory_and_unpack

Differential Revision: D40927291



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

digantdesai reviewed (approved):
LGTM

jianyuh reviewed (changes requested):
Check lint / test failures.

kwanghoon-meta commented:
@jianyuh 
https://github.com/pytorch/pytorch/actions/runs/3978787028/jobs/6828556833
I'm unsure if this cancelled job is related to my PR.
Would you please check?

jianyuh reviewed (approved):
LGTM! cc @jerryzh168 : who is the current maintainer for QNNPACK?

digantdesai reviewed (approved):
LGTM

kwanghoon-meta committed:
[ FL-gradient quantization] Adding QNN unpack feature (#92714)

Summary:
We are trying to add a new feature for quantized gradient computation which enables backward() function for QNNPACK

Pull Request resolved: https://github.com/pytorch/pytorch/pull/92714

Test Plan:
buck2 test //caffe2/test/quantization:quantization -- test_qlinear_qnnpack_free_memory_and_unpack

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

Reviewed By: digantdesai, jianyuh

Differential Revision: D40927291

Pulled By: kwanghoon-meta

fbshipit-source-id: 50e7330c1746655261fa00e41a4e4c966c27bb78
